crates/graphql/src/schema.rs

```diff
@@ -5,13 +5,92 @@ use dataloader::{non_cached::Loader, BatchFn};
 use indexer_core::{
     db::{
         models,
-        tables::{metadata_creators, metadata_jsons, metadatas, storefronts},
+        tables::{attributes, metadata_creators, metadata_jsons, metadatas, storefronts},
         Pool,
     },
     prelude::*,
 };
 use juniper::{EmptyMutation, EmptySubscription, GraphQLObject, RootNode};
 
+#[derive(Debug, Clone)]
+struct Creator {
+    address: String,
+}
+
+#[juniper::graphql_object(Context = AppContext)]
+impl Creator {
+    fn address(&self) -> String {
+        self.address.clone()
+    }
+
+    pub fn attribute_groups(&self, context: &AppContext) -> Vec<Property> {
+        let conn = context.db_pool.get().unwrap();
+
+        let metadatas: Vec<String> = metadata_creators::table
+            .select(metadata_creators::metadata_address)
+            .filter(metadata_creators::creator_address.eq(self.address.clone()))
+            .load(&conn)
+            .unwrap();
+
+        let metadata_attributes: Vec<models::MetadataAttribute> = attributes::table
+            .select(attributes::all_columns)
+            .filter(attributes::metadata_address.eq(any(metadatas)))
+            .load(&conn)
+            .unwrap();
+
+        let attribute_group_lookup =
+            metadata_attributes
+                .into_iter()
+                .fold(HashMap::new(), |mut acc, attribute| {
+                    let trait_type_name = attribute
+                        .trait_type
+                        .map_or_else(String::new, Cow::into_owned);
+                    let value_name = attribute.value.map_or_else(String::new, Cow::into_owned);
+
+                    let trait_type = acc.entry(trait_type_name).or_insert_with(HashMap::new);
+
+                    let value_count = trait_type.entry(value_name.clone()).or_insert(0);
+                    let next_value_count = *value_count + 1;
+
+                    trait_type.insert(value_name, next_value_count);
+
+                    acc
+                });
+
+        let mut attribute_groups: Vec<Property> = vec![];
+
+        for (attribute_group_name, attribute_group) in attribute_group_lookup {
+            let mut variants: Vec<PropertyVariant> = vec![];
+
+            for (variant_name, count) in attribute_group {
+                variants.push(PropertyVariant {
+                    name: variant_name,
+                    count,
+                });
+            }
+
+            attribute_groups.push(Property {
+                name: attribute_group_name,
+                variants,
+            });
+        }
+
+        attribute_groups
```

Comment:
My ideal for this is to have database views that recalculated based on computed columns but this should get us solid traction. Can cache the queries with a ttl as the next improvement.

Reply:
I opted for commuting with rust vs making a bunch of select where count queries within sql.

Reply:
Yeah, diesel only really supports collating rows outside of SQL.

Reply:
@kespinola can i get a rundown on what this code is doing?

Reply:
@ray-kast its taking all the attributes for the metadata associated to the creator and reducing them to a map that looks like this
```
{"Size": {"Small (S)": 14, "Extra Large (XL)": 4, "Medium (M)": 9}, "Type": {"Land": 44}, "Light Source": {"Yes": 28}, "Edition": {"Founders Edition (FE)": 11, "Founders Edition": 5}, "Dynamic Bones": {"Yes": 28}, "Height": {"50m ": 1, "25n": 1, "25m": 33, "50m": 8, "25": 1}, "Lip Sync": {"Yes": 28}, "MBs": {"100": 7, "200": 4, "75": 7, "600": 4, "250": 5}, "Custom Shaders": {"Yes": 28}, "Full Body Tracking": {"Yes": 28}, "Size m²": {"600": 35, "1500": 9}, "Finger Tracking": {"Yes": 28}, "Parcel Size": {"Medium (M)": 35, "Extra Large (XL)": 9}}
```
Then transforming into a list structure for the ui which is more usable but less efficient for doing the counting.